portfolio/src/main/webapp/script.js

```diff
@@ -208,12 +208,35 @@ function checkLogin() {
   });
 }
 
+let markers = [];
```

Comment:
This is a "global variable" which isn't great, but with the current structure of the application it isn't real clear how to fix that.

One option I have in mind makes it a bit better, but might be onerous, so [optional].

Create a class "Map", with constructor accepting `map` and with functions:
* `createMarker(latLng, windowText)`: creates the marker and infowindow and stores them locally.
* `clearMarkers()`: Erases all markers and closes windows.

Reply:
(after creating the class, create a singleton global variable when you're first creating the map)